pkg/cli/filesystem/filesystem.go

```diff
@@ -0,0 +1,15 @@
+package filesystem
+
+import (
+	"io/fs"
+)
+
+// FileSystem is an interface that defines the methods needed to interact with a file system.
```

Comment:
Could you explain the use cases for OSFS and MemMapFS? It seems like MemMapFS is only used for testing while OSFS is used when generating the manifest files

Reply:
That's correct. In the real world (when a user runs the command), they expect to have the file generated for them on their OS filesystem (so we use OSFS). Since we don't want to have a whole OS filesystem for unit tests we can just use the MemMapFS.